Core/src/RequestCallerTrait.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Copyright 2023 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Core;
+
+use Google\Cloud\Core\ArrayTrait;
+use Google\Cloud\Core\Exception\NotFoundException;
+use Google\Cloud\Core\Exception\ServiceException;
+use Google\Cloud\Core\TimeTrait;
+use Google\Cloud\Core\WhitelistTrait;
+use Google\Protobuf\NullValue;
+
+/**
+ * Provides shared functionality for making requests to GAPIC clients
+ * agnostic of the protocol.
+ */
+trait RequestCallerTrait
```

Comment:
The methods in this trait currently reside in `GrpcTrait` and has been extracted into a trait for consistency. We can easily just keep these in `RequestHandler` if that feels more intuitive and less confusing.